### ISSUE TITLE
editing cursor isn't not correct positionned for a blind user

### ISSUE THREAD

**Issue by ignacevdv** [ajaxorg/ace#3012]
Hello, I'm a blind user of windows 7 and I use a screenreader JAWS from Freedom Scientific and I want to use Codebender to program the Arduino.
The editing field on this website is using ACE and the members of Codebender asks my to write here my problem.
When I want to edit the code on the field on this site, everytime the cursor isn't on the perfect position to edit or adding code into this field.
Please could you take a look for this?
There are many blind users that want to use Codebender but when this problem continues we are no longer interessed in ACE and the platforms that use ACE.
I was already trying IE V11, Firefox and Google Chrome but not works at all.
Hope you can help us asap.
Many thanks in advance and have  a nice day!
Kind regards,
Ignace

**Comment by MB-Hilo** [comment]
Hi,
Random programmer here browsing the open issues. I took a look at how to solve this. Though it looks like it is more of a feature request than a quick fix. My suggestion would be to copy and paste the code from Ace editor in the browser to a code editor like Notepad++. Hope you go far with your Arduino.

**Comment by ignacevdv** [comment]
Hello Marc, Many thanks for your answere. Actually I'm using already this work-arround but it is  verry hard to work with this W.A.. Every time I must change 1 or more characters into the code I must follow the several steps to have succes. And, it is verry  difficult to positioning the cursor again into the editfield on Codebender.
I hope you'll proviede a workfriendly solution for us as completely blind programmers.
Many thanks in advance!

**Comment by ignacevdv** [comment]
Hello Marc, 

 

Many thanks for your answere. Actually I'm using already this work-arround but it is  verry hard to work with this W.A.. Every time I must change 1 or more characters into the code I must follow the several steps to have succes. And, it is verry  difficult to positioning the cursor again into the editfield on Codebender.

I hope you'll proviede a workfriendly solution for us as completely blind programmers.

Many thanks in advance!

 

Ignace

Van: Mark Bradley [mailto:notifications@github.com] 
Verzonden: dinsdag 12 juli 2016 6:16
Aan: ajaxorg/ace
CC: ignacevdv; Author
Onderwerp: Re: [ajaxorg/ace] editing cursor isn't not correct positionned for a blind user (#3012)

 

Hi,
Random programmer here browsing the open issues. I took a look at how to solve this. Though it looks like it is more of a feature request than a quick fix. My suggestion would be to copy and paste the code from Ace editor in the browser to a code editor like Notepad++. Hope you go far with your Arduino.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/ajaxorg/ace/issues/3012#issuecomment-231933164> , or mute the thread <https://github.com/notifications/unsubscribe/ATPt_MadMe9nYiTvu0qF59IkbjyYSVmgks5qUxULgaJpZM4I_xzP> .Afbeelding verwijderd door afzender.

No virus found in this message.
Checked by AVG - www.avg.com
Version: 2016.0.7640 / Virus Database: 4627/12599 - Release Date: 07/11/16

**Comment by InspiredGuy** [comment]
I have tried to reproduce the issue on https://edu.codebender.cc/class_demo using VoiceOver on Mac and Narrator on Windows 10, but wasn't able to. https://edu.codebender.cc/class_demo seems to be using ace v1.2.5. We have recently made improvements to Ace accessibility, so our advice for codebender team would be to update to latest version and to use `enableKeyboardAccessibility` option of the editor, which was introduced recently. It prevents editor from trapping the keyboard and has other improvements, so it would be useful in any case.
